src/config.ts

```diff
@@ -19,11 +19,30 @@ declare global {
 }
 
 export function getTargetOrigin() {
-  return window.__tbxConfig?.targetOrigin || ".testbox.com";
+  return getConfigItem("targetOrigin") || ".testbox.com";
 }
 
 export function getLogLevel() {
-  return window.__tbxConfig.logLevel || "none";
+  return getConfigItem("logLevel") || "none";
+}
+
+export function setConfig<K extends keyof TestBoxConfig>(
+  config?: TestBoxConfig
+) {
+  window.__tbxConfig = config || {};
+  return window.__tbxConfig;
+}
+
+export function setConfigItem<K extends keyof TestBoxConfig>(
+  key: K,
+  value: TestBoxConfig[K]
+) {
+  if (!window.__tbxConfig) {
+    console.error("TestBox configuration undefined!");
```

Comment:
❓ Is there a scenario where we'd like to just set a config item even if there is no initial config loaded? Not sure there is, just wondering.

⛏️ Also, can you use the logging we defined so that it will only log out on the appropriate error level? See `src/utils/logging.ts`

Reply:
hmm potentially. But this scenario should never happen now. It's just a check to make typescript happy